### PR TITLE
WIP Dynamic service methods in functional tests

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/grpc/grpc-dotnet/issues/80

Also adds a `MapGrpcService` that allows explicit binding. This is for code gen services that place BindService in a different class structure.

### CODE REVIEW

File: src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs [hunk 1]
```diff
@@ -62,6 +62,7 @@ public static IGrpcServerBuilder AddGrpc(this IServiceCollection services)
 
             services.AddRouting();
             services.AddOptions();
+            services.AddLogging();
```
Comment (davidfowl) [inline src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs]: For easy testing? Nothing else in our stack does this.

Comment (JamesNK) [reply src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs]: I noticed AddGrpc by itself errors. Is that fine?

Comment (davidfowl) [reply src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs]: Logging is added by the host, why does it error?

Comment (JamesNK) [reply src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs]: It errored by itself in a test. If every other framework leaves it to the host then I'll remove it.

**Comment by JamesNK** [comment]
Feel free to review but I might change this up a bit. I'm going to try out a couple more things to remove the hard dependency services have on `typeof(Service).GetMethod(Method.Name)`.

**Pull request closed without merging**